src/display.c

```diff
@@ -244,14 +245,23 @@ static uintptr_t display_map_fb(uintptr_t iova, u64 paddr, u64 size)
 
 const display_config_t *display_get_config(void)
 {
+    const display_config_t *conf = NULL;
+
     if (adt_is_compatible(adt, 0, "J473AP"))
-        return &display_config_m2;
+        conf = &display_config_m2;
     else if (adt_is_compatible(adt, 0, "J474sAP") || adt_is_compatible(adt, 0, "J475cAP"))
-        return &display_config_m2_pro_max;
+        conf = &display_config_m2_pro_max;
     else if (adt_is_compatible(adt, 0, "J180dAP") || adt_is_compatible(adt, 0, "J475dAP"))
-        return &display_config_m2_ultra;
+        conf = &display_config_m2_ultra;
     else
-        return &display_config_m1;
+        conf = &display_config_m1;
+
+    has_dcp = !!adt_path_offset(adt, conf->dcp);
```

Comment:
`adt_path_offset()` returns -1 on failure, so this breaks the has_dcp check.
```suggestion
    has_dcp = adt_path_offset(adt, conf->dcp) > 0;
```